### ISSUE TITLE
code-block directive with same :caption: causes warning of duplicate target

### ISSUE THREAD

**Issue by jason-s** [sphinx-doc/sphinx#2931]
test1.rst:

```
Yippee
======

+---------------------------------+------------------------------------+
| foo                             | bar                                |
+=================================+====================================+
| .. code-block:: c               | .. code-block:: c                  |
|    :caption: m2.h               |    :caption: m2.h                  |
|                                 |                                    |
|    blah blah blah foo           |    #include <stdint.h>             |
|                                 |    blah blah blah bar              |
+---------------------------------+------------------------------------+
```

I get this message: `test1.rst:2: WARNING: Duplicate explicit target name: "m2.h".`

If I change one of the m2.h's to m1.h, the warning goes away.

Looks like `code-block` is creating a target with the caption name, not sure why it is doing this.
Multiple code-blocks with the same caption should be allowed.

**Comment by jason-s** [comment]
Looks like if I specify a `:name:` argument then I can prevent the warning.

```
Yippee
======

+---------------------------------+------------------------------------+
| foo                             | bar                                |
+=================================+====================================+
| .. code-block:: c               | .. code-block:: c                  |
|    :caption: m2.h               |    :caption: m2.h                  |
|    :name: m2_foo                |    :name: m2_bar                   |
|                                 |                                    |
|    blah blah blah foo           |    #include <stdint.h>             |
|                                 |    blah blah blah bar              |
+---------------------------------+------------------------------------+
```

**Comment by tk0miya** [comment]
Sorry for inconvenience. It's bug of Sphinx.

Note: This comes from #1742. It diverts `caption` option to `name` if `name` not given.

**Comment by tk0miya** [comment]
Fixed at 82c153f (stable).
It will be released as 1.4.7.

Thank you for reporting!
